src/main/java/org/opentripplanner/model/TimetableSnapshot.java

```diff
@@ -369,4 +385,19 @@ public String toString() {
         return timetables.keySet();
     }
 
+    private void addPatternToIndex(TripPattern tripPattern) {
+        for (Stop stop: tripPattern.getStops()) {
+            if (!patternsForStop.containsEntry(stop, tripPattern)) {
```

Comment:
This conditional is probably not necessary, even from a performance point of view. The meaning might be clearer with only the put() call and no conditional.